sdk/src/client/api/address.rs

```diff
@@ -82,8 +56,8 @@ impl<'a> GetAddressesBuilder<'a> {
     }
 
     /// Set bech32 human readable part (hrp)
-    pub fn with_bech32_hrp(mut self, bech32_hrp: impl Into<Option<Hrp>>) -> Self {
-        self.bech32_hrp = bech32_hrp.into();
+    pub fn with_bech32_hrp(mut self, bech32_hrp: impl HrpLike) -> Self {
+        self.bech32_hrp = bech32_hrp.to_hrp_unchecked();
```

Comment:
Do we really not want to check here?

Reply:
I thought about it. It would make this API less clean. I could separate this into two methods of course but I felt like it really wasn't necessary. I dunno though, if you feel strongly let me know.

Reply:
It just feels a bit inconsistent with what you've been doing so far, this should at least have an _unchecked suffix. I don't think it wouldn't be clean to have a with_ that returns an error though, that wouldn't be the first.